modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2DeveloperKeysForDXP.testcase

```diff
@@ -121,4 +121,266 @@ definition {
+	@ignore = "true"
+	@priority = "4"
+	test UserPartnerMemberAndRequesterRoleCannotSeeSomeActionsForDXP {
+
+		// Needs refactor since it depends on example Avangrid Portals account
+
+		task ("Go to CP2.0 and join a project") {
+			ApplicationsMenu.gotoSite(key_site = "Customer Portal Site");
+
+			CustomerPortal.enterOnProject(title = "${testAccount}");
+		}
+
+		task ("Go to the product activation page") {
+			CP2NavigationMenu.goToProductActivation();
+
+			CP2NavigationMenu.clickOnProductButton(product = "${testProduct}");
+		}
+
+		task ("Click in the Action button and assert the user with the right role can see the option") {
+			CustomerPortal.clickAtButton(text = "Actions");
+
+			AssertVisible(
+				key_action = "Export All Key Details (csv)",
+				locator1 = "CP2ActivationKeys#ACTIONS_DROPDOWN");
+		}
+
+		task ("Given a user signs in on OKTA to update the Customer Portal") {
+			CustomerPortal.loginOktaToUpdateCP();
+
+			Navigator.openURL();
+
+			ApplicationsMenu.gotoSite(key_site = "${testSiteName}");
+		}
+
+		task ("Change the user role in the project") {
+			Account.openAccountsAdmin();
+
+			CustomerPortal.accountTableProject(title = "NASA ICE");
+
+			Account.addAssigneeToRoleViaUsersTab(
+				accountRoleTitle = "Partner Member",
+				accountUserName = "Test Test");
+
+			Account.addAssigneeToRoleViaUsersTab(
+				accountRoleTitle = "Requester",
+				accountUserName = "Test Test");
+		}
+
+		task ("Go to CP2.0 and join a project") {
+			ApplicationsMenu.gotoSite(key_site = "Customer Portal Site");
+
+			Refresh();
+
+			CustomerPortal.enterOnProject(title = "${testAccount}");
+		}
+
+		task ("Go to the product activation page") {
+			CP2NavigationMenu.goToProductActivation();
+
+			CP2NavigationMenu.clickOnProductButton(product = "${testProduct}");
+		}
+
+		task ("Click in the Action button and assert if the user with the right role can see the options") {
+			CustomerPortal.clickAtButton(text = "Actions");
+
+			AssertElementNotPresent(
+				key_text = "Generate New",
+				locator1 = "CP2ActivationKeys#ACTIONS_DROPDOWN");
+
+			CustomerPortal.clickAtButton(text = "Actions");
+
+			AssertElementNotPresent(
+				key_text = "Deactivate",
+				locator1 = "CP2ActivationKeys#ACTIONS_DROPDOWN");
+		}
+
+		task ("Change the user role in the project") {
+			Account.openAccountsAdmin();
+
+			CustomerPortal.accountTableProject(title = "NASA ICE");
+
+			Account.unassignUserRole(
+				accountRoleTitle = "Requester",
+				accountUserName = "Test Test");
+
+			Account.unassignUserRole(
+				accountRoleTitle = "Partner Member",
+				accountUserName = "Test Test");
+		}
+	}
+
+	@description = "Automation ID: LPS-166940 | Test Summary: Verify that users with the roles Partner Members, Requesters and Users cannot see the actions (Generate New - Deactivate - Actions Menu - Deactivate button)."
```

Comment:
What is the difference between this test and the test above? ( LPS-166937 )
Aren't they with the same purpose?